### DISCUSSION TITLE
How to detect URI on websocket connect

### DISCUSSION THREAD

**Discussion by be-thomas** [uNetworking/uWebSockets.js#504] [Q&A]
I need a way to authenticate users. I want to disconnect a user immediately on connect based on query parameters.

This can be done in node express-ws library, it can be done by handling the on-upgrade handler -

[https://stackoverflow.com/questions/59070321/websockets-token-authentication-using-middleware-and-express-in-node-js/59314360#59314360](Node Websocket Auth).

A code snippet would be great

**Comment by hst-m** [comment]
https://github.com/uNetworking/uWebSockets.js/discussions/339#discussioncomment-172632

**Comment by ramiroaisen** [comment]
You just have to pass the `upgrade(res, req, ctx)` function to `app.ws()`
